src/Travers.php

```diff
@@ -65,6 +89,38 @@ public function find($key)
         return $this->parseParam($this->parseKeys($key), $this->tree);
     }
 
+    /**
+     * @param string $key
+     * @param array  $tree
+     *
+     * @return array|mixed|null
+     * @throws BranchNotFoundException
+     */
+    protected function removeParam($key, array $tree)
```

Comment:
Let's keep this function similar to setParam & parseParam, and pass $this->parseKeys($key) as the first arg.
```
$this->setTree($this->removeParam($this->parseKeys($path), $this->tree));
...
protected function removeParam(array $keys, array $tree)
```